src/test/java/ch/uzh/ifi/hase/soprafs21/service/PathServiceIntegrationTest.java

```diff
@@ -0,0 +1,111 @@
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.context.SecurityContext;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.test.annotation.DirtiesContext;
+import org.springframework.test.context.jdbc.Sql;
+import org.springframework.web.server.ResponseStatusException;
+import java.util.Optional;
+import static org.junit.jupiter.api.Assertions.*;
+
+@SpringBootTest
+@Sql(value = {"/data_init.sql"})
+@DirtiesContext(classMode = DirtiesContext.ClassMode.BEFORE_EACH_TEST_METHOD)
+public class PathServiceIntegrationTest {
+
+    @Autowired
+    private UserRepository userRepository;
+
+    @Mock
+    private PathRepository pathRepository;
+
+    @Autowired
+    private GeometryFactory geometryFactory;
+
+    @Autowired
+    private PathService pathService;
+
+    @Mock
+    private Authentication authenticationMock;
+
+    @Mock
+    private SecurityContext securityContextMock;
+
+    private Path path;
+
+    private Path path_1;
+
+    @BeforeEach
+    void setUp() {
+        UserOverviewDto userOverviewDto = new UserOverviewDto();
+        userOverviewDto.setEmail("mark@twen.de");
+        userOverviewDto.setId(1L);
+        userOverviewDto.setName("mark");
+        Mockito.when(authenticationMock.getPrincipal()).thenReturn(userOverviewDto);
+        Mockito.when(securityContextMock.getAuthentication()).thenReturn(authenticationMock);
+        SecurityContextHolder.setContext(securityContextMock);
+
+        Coordinate [] coordinates = new Coordinate[5];
+        coordinates[0] = new Coordinate(8.432962, 47.378622);
+        coordinates[1] = new Coordinate(8.474358, 47.373311);
+        coordinates[2] = new Coordinate( 8.470429, 47.357797);
+        coordinates[3] = new Coordinate(8.437273, 47.362384);
+        coordinates[4] = new Coordinate(8.432962, 47.378622);
+        LineString routeCoordinates = geometryFactory.createLineString(coordinates);
+        path = Path.builder().route(routeCoordinates).creator(userRepository.getOne(1L)).distance(2.0).build();
+        path_1 = Path.builder().route(routeCoordinates).creator(userRepository.getOne(2L)).distance(2.0).build();
+    }
+
+    @Test
+    void testAddPath() {
+        Path persistedPath = pathService.savePath(path);
+        assertEquals(path.getCreator().getId(), persistedPath.getCreator().getId());
+        assertEquals(path.getDistance(), persistedPath.getDistance());
+    }
+
+    @Test
+    void testDeletePath(){
+        Path persistedPath = pathService.savePath(path);
+        this.pathService.deletePath(persistedPath.getId());
+        assertEquals(Optional.empty(), pathRepository.findById(persistedPath.getId()));
+    }
+
+    @Test
+    void testDeletePath_withUnauthenticateUser(){
+
+        Path persistedPath = pathService.savePath(path_1);
```

Comment:
This should throw an exception because the authenticated user is different from the creator. But it does not throw any exceptions because you check it in the controller on adding the path. Please, move the check to the Service.

Reply:
This test is for checking the delete path scenario and its just an integration test which is calling service layer directly and it will not be called directly for any application scenario.

Reply:
This integration test shows that it does not detect a particular scenario. And this is because there is no check for authentication.

Reply:
the check is at the controller layer, and I do not see the need to move it to service.